src/pages/tBTC/Bridge/components/BridgeProcessCardTitle.tsx

```diff
@@ -1,50 +1,81 @@
-import { FC } from "react"
-import { HiArrowNarrowLeft } from "react-icons/all"
-import { Icon, Stack, LabelSm } from "@threshold-network/components"
-import { tBTCFillBlack } from "../../../../static/icons/tBTCFillBlack"
-import { BridgeProcess, MintingStep } from "../../../../types/tbtc"
-import { TBTCText } from "../../../../components/tBTC"
+import { ComponentProps, FC, ReactNode } from "react"
+import {
+  BodyLg,
+  BodySm,
+  Box,
+  Flex,
+  HStack,
+  Icon,
+  VisuallyHidden,
+  VStack,
+} from "@threshold-network/components"
+import { MintingStep, MintingSteps as mintingSteps } from "../../../../types"
+import { FaChevronLeft as ChevronLeftIcon } from "react-icons/fa"
 
-const bridgeProcessToText: Record<BridgeProcess, string> = {
-  mint: "minting process",
-  unmint: "unminting process",
-}
-
-type CommonProps = {
-  bridgeProcess?: BridgeProcess
-}
-
-type ConditionalProps =
-  | {
-      previousStep?: MintingStep
-      onPreviousStepClick: (previosuStep: MintingStep) => void
-    }
-  | {
-      previousStep?: never
-      onPreviousStepClick?: never
-    }
-
-type Props = CommonProps & ConditionalProps
+const TOTAL_STEPS = mintingSteps.length - 1
+// -1 because we don't count the success step
 
-export const BridgeProcessCardTitle: FC<Props> = ({
-  previousStep,
+export const BridgeProcessCardTitle: FC<
+  {
+    step: number
+    title: string
+    description?: ReactNode
+    afterDescription?: ReactNode
+    onPreviousStepClick?: (previousStep?: MintingStep) => void
+  } & ComponentProps<typeof BodyLg>
+> = ({
+  step,
+  title,
+  description,
+  afterDescription,
   onPreviousStepClick,
-  bridgeProcess = "mint",
+  ...restProps
 }) => {
   return (
-    <Stack direction="row" mb={8} align={"center"}>
-      {previousStep && (
-        <Icon
-          cursor="pointer"
-          onClick={() => onPreviousStepClick(previousStep)}
-          mt="4px"
-          as={HiArrowNarrowLeft}
-        />
-      )}
-      <Icon boxSize="32px" as={tBTCFillBlack} />
-      <LabelSm>
-        <TBTCText /> - {bridgeProcessToText[bridgeProcess]}
-      </LabelSm>
-    </Stack>
+    <VStack align="start" spacing={6} {...restProps}>
+      <HStack spacing={4}>
+        {onPreviousStepClick ? (
+          <Flex
+            as={"button"}
+            onClick={() => onPreviousStepClick()}
+            align="center"
+            border="1px solid #333"
+            minW="auto"
+            h="auto"
+            rounded="md"
+            p={2}
+          >
+            <Icon
+              as={ChevronLeftIcon}
+              w={4}
+              h={4}
+              color="hsl(182, 100%, 70%)"
+            />
+            <VisuallyHidden>Previous step</VisuallyHidden>
+          </Flex>
+        ) : null}
+        <BodySm
+          rounded="md"
+          bg="#1E1E1E"
+          p={2}
+          fontWeight="medium"
+          lineHeight={4}
+          color="hsl(182, 100%, 70%)"
+        >
+          {step}/{TOTAL_STEPS}
+        </BodySm>
```

Comment:
I think we can use `Badge` here. It's so similar to `Action off-chain` badge below, that I think it would make more sense:

```suggestion
        <Badge variant="subtle" rounded="md" p={"2"} lineHeight={4}>
          {step}/{TOTAL_STEPS}
        </Badge>
```

Reply:
Ref commit: 5e281add